### DISCUSSION TITLE
Is it possible to define the search pattern using wildcards?

### DISCUSSION THREAD

**Discussion by raindropsfromsky** [Flow-Launcher/Flow.Launcher#1176] [Ideas]
Usually FL shows far too many results. Although it highlights the matching pattern in each entry, still the user has to read through all entries.

It would be much easier if the user can define his search more precisely using wildcards like * and ?.
It would help even further if he can define the desired file extension in the search pattern.
This would filter out a lot of non-matching results.

**Comment by taooceros** [comment]
Currently the majority results of Flow come from two sources.
1. Program index, which use a partial string match to search for results, and does not support wildcards. I will suspect the hardness of implementing wildcard since the matching is not regex based.
2. Windows Search | Everything Search. They both support wildcards.

**Comment by raindropsfromsky** [comment]
I tried this, and found that when I use wildcards, the matched patterns are not shown in a different color.
It would be nice if the pattern that matches the wildcard-based search patterns are highlighted (if possible, with a different color than an exact match).

Really speaking, these "near matches" are (much) more difficult to spot than the exact matches, because they may be different in each case! So if it is good to highlight the exact matches, then it is even more valuable if the near matches are pointed out in the results!

**Reply by jjw24** [reply]
If I remember correctly, wildcard searches assuming you are referring to the uses of '>' or '*', are executed with Microsoft's search library which does not offer the same flexibility as our own search to allow for highlighting.

**Comment by Muffexx** [comment]
This might help: https://github.com/Flow-Launcher/Flow.Launcher/issues/1717
tl;dr: use regex by typing: "regex:*yourExpression*"
